### PR TITLE
Prepare: Update core function headers for refactor of GUI rendering

### PR DESCRIPTION
**Pull request by techgeeknz**
This is split off from https://github.com/OpenTTD/OpenTTD/pull/8217; part 1 of 3.

### CODE REVIEW

File: src/core/math_func.hpp [hunk 1]
```diff
@@ -308,91 +308,83 @@ int DivideApprox(int a, int b);
...
- * @param b Denominator
- * @return Quotient, rounded towards negative infinity
+ * Computes a / b rounded towards negative infinity for integer a and positive integer b.
+ * @param a Numerator.
+ * @param b Denominator.
+ * @return Quotient, rounded towards negative infinity.
  */
-template <typename S, typename U>
-static inline S DivTowardsNegativeInf(S a, U b)
+static inline int DivTowardsNegativeInf(const int a, const uint b)
```
Comment (nielsmh) [inline src/core/math_func.hpp:372]: You add these functions in the first commit, then change them in a later?

File: src/core/math_func.hpp [hunk 2]
```diff
@@ -308,91 +308,83 @@ int DivideApprox(int a, int b);
...
+ * Computes round(a / b) for integer a and positive integer b.
+ * @param a Numerator.
+ * @param b Denominator.
+ * @return Quotient, rounded to nearest.
  */
-static inline int RoundDivSU(int a, uint b)
+static inline int RoundDivSU(const int a, const uint b)
 {
-	const int _b = static_cast<int>(b);
+	const int b_ = static_cast<int>(b);
```
Comment (nielsmh) [inline src/core/math_func.hpp:339]: Why not just name this variable right in the commit adding it?

Also, an idea: Check if `b >= std::numeric_limits<int>::max` and return 0 if it is. That avoids UB casting b to a type that can't fit the value, and will still have the correct result.

**Comment by LordAro** [comment]
Commits should be made up of individual logical changes. You're doing that, and that's good. But if you can reorder things so that commits are smaller or never necessary at all, that's even better.

And again, you're adding functions that are then unused. These should be added (as a separate commit) in the PR that requires them

**Comment by techgeeknz** [comment]
> And again, you're adding functions that are then unused. These should be added (as a separate commit) in the PR that requires them

This is splitting hairs. These are used in the follow-up PR; you asked me to split into smaller, easier-to-review chunks, and I've done that. Now you want them merged back together?!

File: src/core/math_func.hpp (hunk 1 above)
Comment (techgeeknz) [reply src/core/math_func.hpp:372]: Technically, @JGRennison added the functions and I changed them. Either way, I thought I'd squashed those together.

File: src/core/math_func.hpp (hunk 2 above)
Comment (techgeeknz) [reply src/core/math_func.hpp:339]: I wish I'd known about _that_ one earlier 😉.

**Comment by LordAro** [comment]
I want you to do relevant things at the same time. Adding functions in a separate set of changes and then not using them is pointless - it may be weeks or months before the main body of work is used (if indeed it ever ends up happening at all).

The only thing I said should be split out is the Viewport renaming, which is otherwise irrelevant to the rest of the changes and causes a lot of "diff bloat" that makes the actual relevant changes harder to review.

Splitting out the stdafx reformatting & adding sanity checks to existing functions is also recommended to be a separate PR - they are changes that aren't required for the main body of work.

**Comment by techgeeknz** [comment]
> Splitting out the stdafx reformatting & adding sanity checks to existing functions is also recommended to be a separate PR - they are changes that aren't required for the main body of work.

That is what I thought we were doing here; but never mind, we'll ride the merry-go-round again next week.

**Pull request closed without merging**